plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/job/OpenShiftJobs.java

```diff
@@ -49,18 +55,43 @@ protected IStatus doRun(IProgressMonitor monitor) {
 				List<IProject> oldProjects = connection.getResources(ResourceKind.PROJECT);
 				IStatus status = super.doRun(monitor);
 				if(status.isOK()) {
-					List<IProject> newProjects = new ArrayList<>(oldProjects);
-					newProjects.remove(project);
-					ConnectionsRegistrySingleton.getInstance().fireConnectionChanged(
-							connection , 
-							ConnectionProperties.PROPERTY_PROJECTS, 
-							oldProjects, 
-							newProjects);
+					if(waitForServerToReconcileProjectDelete(connection, project)) {
+						List<IProject> newProjects = new ArrayList<>(oldProjects);
+						newProjects.remove(project);
+						ConnectionsRegistrySingleton.getInstance().fireConnectionChanged(
+								connection , 
+								ConnectionProperties.PROPERTY_PROJECTS, 
+								oldProjects, 
+								newProjects);
+						
+					}
 				}
 				return status;
 			}
+
+			private boolean waitForServerToReconcileProjectDelete(Connection conn, IProject project) {
+				boolean deleted = false;
+				long sleep = 0;
+				do {
+					try {
+						conn.getResource(project);
+						Thread.sleep(PROJECT_DELETE_DELAY);
+						sleep = sleep + PROJECT_DELETE_DELAY;
+					} catch (InterruptedException e1) {
+					} catch(OpenShiftException ex) {
+						if(ex.getStatus() != null) {
```

Comment:
The actual reason for the missing status is an issue in the client lib:

DefaultClient#createOpenShiftException()
```
			if(status.getCode() == STATUS_FORBIDDEN) {
				if(StringUtils.isNotBlank(token)) { //truly forbidden
					return new ResourceForbiddenException(status.getMessage(), e);
```
The resourceForbiddenException that is being constructed simply wont add the status, it only adds the message.